dependency/vault_pki_test.go

```diff
@@ -0,0 +1,184 @@
+// go:build ignore
+
+package dependency
+
+import (
+	"bytes"
+	"encoding/pem"
+	"errors"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/consul-template/renderer"
+	"github.com/hashicorp/vault/api"
+)
+
+func Test_VaultPKI_goodFor(t *testing.T) {
+	_, cert, err := getCert([]byte(testCert))
+	if err != nil {
+		t.Error(err)
+	}
+	dur, ok := goodFor(cert)
+	if ok != false {
+		t.Error("should be false")
+	}
+	// duration should be negative as cert has already expired
+	// but still tests cert time parsing (it'd be 0 if there was an issue)
+	if dur >= 0 {
+		t.Error("cert shouldn't be 0 or positive (old cert)")
+	}
+}
+
+func Test_VaultPKI_getCert(t *testing.T) {
```

Comment:
Fleshed out these test some more with a series of tests using a malformed certificate to validate that it can still find a good cert in a file with a bad one and to validate that it returns errors properly (along with testing it returns that error even if stuff happens after).